### PR TITLE
Implement archive size API (metaArchiveSize).

### PR DESCRIPTION
**Pull request by frankban**

### CODE REVIEW

File: internal/v4/api_test.go [hunk 1]
```diff
@@ -94,11 +94,30 @@ var metaEndpoints = []metaEndpoint{{
...
+	checkURL:        "cs:bundle/wordpress-42",
+	assertCheckData: sizeChecker("cs:bundle/wordpress-42"),
 }}
 
+func sizeChecker(id string) func(c *gc.C, store *charmstore.Store, data interface{}) {
+	return func(c *gc.C, store *charmstore.Store, data interface{}) {
+		var doc mongodoc.Entity
+		err := store.DB.Entities().FindId(id).One(&doc)
+		c.Assert(err, gc.IsNil)
+		c.Assert(data, gc.Equals, doc.Size)
```
Comment (rogpeppe) [inline internal/v4/api_test.go]: perhaps, for sanity checking:

c.Assert(doc.Size, gc.Not(gc.Equals), int64(0))

too?

**Comment by jrwren** [comment]
LGTM

File: internal/v4/api_test.go (hunk 1 above)
Comment (frankban) [inline internal/v4/api_test.go]: Done.

File: internal/v4/api_test.go [hunk 2]
```diff
@@ -94,11 +94,30 @@ var metaEndpoints = []metaEndpoint{{
...
+	checkURL:        "cs:precise/wordpress-23",
+	assertCheckData: sizeChecker("cs:precise/wordpress-23"),
+}, {
+	name:            "archive-size", // Bundle size.
+	get:             entityFieldGetter("Size"),
+	checkURL:        "cs:bundle/wordpress-42",
+	assertCheckData: sizeChecker("cs:bundle/wordpress-42"),
 }}
 
+func sizeChecker(id string) func(c *gc.C, store *charmstore.Store, data interface{}) {
```
Comment (rogpeppe) [inline internal/v4/api_test.go]: I don't think this is checking what we want it to. the assertCheckData functions are
for sanity checking the get function. By getting the data from the store, we're
sanity checking something against itself - that's why we don't pass the
store to the sanity checking functions.

I think we should lose the newly added *charmstore.Store argument and just
do:

```
func sizeChecker(c *gc.C, data interface{}) {
   c.Assert(data.(int64), gc.Not(gc.Equals), int64(0))
}
```

If we're getting something that's the right type, and it's not zero,
then I'm reasonably happy that the entityFieldGetter("Size")
call is doing the right thing.

**Comment by rogpeppe** [comment]
LGTM modulo the above suggestion.

File: internal/v4/api_test.go (hunk 2 above)
Comment (frankban) [inline internal/v4/api_test.go]: Done.

**Comment by frankban** [comment]
Thanks for the reviews!
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-merge

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results: http://ci.jujugui.org:8080/job/charmstore/58/

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results: http://ci.jujugui.org:8080/job/charmstore/60/

**Comment by jujugui** [comment]
Build failed: Attempt to land pull request failed
        build url: http://ci.jujugui.org:8080/job/charmstore-merge/43

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results: http://ci.jujugui.org:8080/job/charmstore/64/

**Pull request merged**